.github/workflows/common.yml

```diff
@@ -24,9 +24,15 @@ on:
         default: "3.24.x"
         required: false
         type: string
+      build-type:
+        description: "One of the {Debug, Release, MinSizeRel, RelWithDebInfo} (or maybe custom) build types"
+        default: "Debug"
+        required: false
+        type: string
 
 env:
-  CMAKE_MULTIBUILD_CONFIG: "${{ startsWith(inputs.os, 'windows') && '--config Debug' || '' }}"
+  CMAKE_MULTIBUILD_CONFIG: "${{ startsWith(inputs.os, 'windows') && format('--config {0}', inputs.build-type) || '' }}"
+  CMAKE_UNIX_BUILD_TYPE: "${{ startsWith(inputs.os, 'windows') && '' || format('-DCMAKE_BUILD_TYPE={0}', inputs.build-type)}}"
```

Comment:
I would rename `CMAKE_UNIX_BUILD_TYPE` to `CMAKE_BUILD_TYPE_VARIABLE` and `CMAKE_MULTIBUILD_CONFIG` to `CMAKE_CONFIG_ARGUMENT`

Reply:
Done